### ISSUE TITLE
postpone serializing IOSource and builtin decls until `builtins.merge`

### ISSUE THREAD

**Issue by aryairani** [unisonweb/unison#1410]
Currently, this happens at codebase initialization time, in `Codebase.hs`:
https://github.com/unisonweb/unison/blob/e7fd115eeb95f5b7e4ee3bfff84945c7807e9f85/parser-typechecker/src/Unison/Codebase.hs#L96-L105 and gets called in a couple of places:

https://github.com/unisonweb/unison/blob/e7fd115eeb95f5b7e4ee3bfff84945c7807e9f85/parser-typechecker/src/Unison/Codebase.hs#L109-L112

https://github.com/unisonweb/unison/blob/e7fd115eeb95f5b7e4ee3bfff84945c7807e9f85/parser-typechecker/src/Unison/Codebase/FileCodebase.hs#L156-L171

It doesn't really need to happen before those definitions are referenced in the namespace though, which happens through a `pull` (not our problem) or a `builtins.merge`:
https://github.com/unisonweb/unison/blob/e7fd115eeb95f5b7e4ee3bfff84945c7807e9f85/parser-typechecker/src/Unison/Codebase/Editor/HandleInput.hs#L1567-L1573

~There are a couple of ways to approach this, but instead of adding a new `Command` to either a) add the builtins by calling `Codebase.initializeBuiltinCode` or b) adding a provided `TypecheckedUnisonFile`, before updating the namespace~ 

There is already a command to add a provided `TypecheckedUnisonFile`: https://github.com/unisonweb/unison/blob/e7fd115eeb95f5b7e4ee3bfff84945c7807e9f85/parser-typechecker/src/Unison/Codebase/Editor/Command.hs#L69-L70

So just call that, and inline `Codebase.initializeBuiltinCode` into `MergeBuiltinsI`.

**Comment by aryairani** [comment]
I forgot that the runtime depends on being able to look up `base.Unit`.

**Comment by aryairani** [comment]
Implemented in #1411
